app/controllers/miq_task_controller.rb

```diff
@@ -69,38 +66,22 @@ def jobs
                                                      :headers    => @view.headers})
         page << "miqSparkle(false);"  # Need to turn off sparkle in case original ajax element gets replaced
       end
-    else                      # Came in from non-ajax, just get the jobs
-      get_jobs(tasks_condition(@tasks_options[@tabform]))
     end
   end
 
-  def get_jobs(conditions)
+  def list_jobs
     @lastaction = "jobs"
+    @active_tab = @tabform.split("_").last
 
-    if @tabform == "tasks_1"
-      @layout = "my_tasks"
-      @view, @pages = get_view(Job, :conditions => conditions)  # Get the records (into a view) and the paginator
-      drop_breadcrumb(:name => _("My VM and Container Analysis Tasks"), :url => "/miq_task/index?jobs_tab=tasks")
-
-    elsif @tabform == "tasks_2"
-      # My UI Tasks
-      @layout = "my_ui_tasks"
-      @view, @pages = get_view(MiqTask, :conditions => conditions)  # Get the records (into a view) and the paginator
-      drop_breadcrumb(:name => _("My Other UI Tasks"), :url => "/miq_task/index?jobs_tab=tasks")
-
-    elsif @tabform == "tasks_3" || @tabform == "alltasks_1"
-      @layout = "all_tasks"
-      @view, @pages = get_view(Job, :conditions => conditions)  # Get the records (into a view) and the paginator
-      drop_breadcrumb(:name => _("All VM and Container Analysis Tasks"), :url => "/miq_task/index?jobs_tab=alltasks")
-      @user_names = Job.distinct("userid").pluck("userid").delete_if(&:blank?)
-
-    elsif @tabform == "tasks_4" || @tabform == "alltasks_2"
-      # All UI Tasks
-      @layout = "all_ui_tasks"
-      @view, @pages = get_view(MiqTask, :conditions => conditions)  # Get the records (into a view) and the paginator
-      drop_breadcrumb(:name => _("All Other Tasks"), :url => "/miq_task/index?jobs_tab=alltasks")
-      @user_names = MiqTask.distinct("userid").pluck("userid").delete_if(&:blank?)
+    case @tabform
+    when "tasks_1" then @layout = "my_tasks"
+    when "tasks_2" then @layout = "my_ui_tasks"
+    when "tasks_3", "alltasks_1" then @layout = "all_tasks"
+    when "tasks_4", "alltasks_2" then @layout = "all_ui_tasks"
     end
+
+    @user_names = db_class.distinct("userid").pluck("userid").delete_if(&:blank?) if @active_tab.to_i > 2
```

Comment:
yea, there is a tradeoff on creating tasks vs searching this screen.
And since we create a bunch of tasks that are not in the ui, this may be a hit we don't want.

Do we have another mechanism to pick users?
Since this is just moving code (not creating it), I think fixing this is for another PR

trivial change, loose the parameter to `distinct` - it is not necessary due to the `pluck`

```diff
-@user_names = db_class.distinct("userid").pluck("userid").delete_if(&:blank?) if @active_tab.to_i > 2
+@user_names = db_class.distinct.pluck("userid").delete_if(&:blank?) if @active_tab.to_i > 2
```